pkg/jobcreator/controller.go

```diff
@@ -190,6 +220,56 @@ func (controller *JobCreatorController) Start(ctx context.Context, cm *system.Cl
 
 	return errorChan
 }
+func (controller *JobCreatorController) UpdateModuleAllowlist() error {
+	// URL of the GitHub raw content of the allowlist
+	allowlistUrl := "https://raw.githubusercontent.com/lilypad-tech/module-allowlist/master/allowlist.txt"
+
+	// Create a new HTTP request with a context
+	req, err := httpAlias.NewRequest("GET", allowlistUrl, nil)
```

Comment:
We might want to add a new helper function to the `"github.com/lilypad-tech/lilypad/pkg/http"` package to handle this case and others that need a timeout. We could create a `GetRequestWithTimeout` that takes the necessary request parameters and a timeout argument.

The idea is to keep all of our HTTP communication going through one package. And it's pretty likely we'll need something like this again in the future!

Reply:
I added a GetRequestWithTimeout helper function within the pkg/http/utils.go - HTTP comms are going through one package!

Reply:
Awesome! Thanks ❤️